src/main/java/core/basesyntax/service/shopservice/ShopServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.shopservice;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.strategy.OperationHandler;
+import core.basesyntax.service.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> fruitTransactions) {
```

Comment:
The process method correctly uses the OperationStrategy to obtain the correct OperationHandler for each transaction, which aligns with the checklist requirement to avoid using switch-cases and if-else constructions for operation handling.